### PR TITLE
Show nsfw communities if you are logged in and searching communities

### PR DESCRIPTION
**Pull request by dayinjing**
Fix for https://github.com/LemmyNet/lemmy/issues/2103

If a user is logged in and nsfw is enabled in their profile show nsfw communities in community search.

### CODE REVIEW

**Review by Nutomic — APPROVED** [review]
Tested it, works fine

File: crates/db_views_actor/src/community_view.rs [hunk 1]
```diff
@@ -231,11 +232,16 @@ impl<'a> CommunityQueryBuilder<'a> {
           .then_order_by(community_aggregates::published.desc())
       }
     };
-
-    if !self.show_nsfw.unwrap_or(false) {
+    if !self.show_nsfw.unwrap_or(false) && person_id_join == PersonId(-1) {
```
Comment (dessalines) [inline crates/db_views_actor/src/community_view.rs:235]: There's an `if self.my_person_id.is_some() {` below, these should go in sub-blocks there. Don't do the same check twice if unavoidable.

**Review by dessalines — CHANGES_REQUESTED** [review]

File: crates/db_views_actor/src/community_view.rs (hunk 1 above)
Comment (dayinjing) [reply crates/db_views_actor/src/community_view.rs]: Done

**Comment by dayinjing** [comment]
@dessalines Sorry it took me a few days to get to it but i've made those changes.

**Review by Nutomic — APPROVED** [review]

**Review by dessalines — APPROVED** [review]
Looks good, thx!

**Pull request merged**